org.lflang/src/org/lflang/util/FileUtil.java

```diff
@@ -383,6 +387,88 @@ private static void copyDirectoryFromJar(JarURLConnection connection, final Path
+     * a source directory). This does the work of CMake by explicitly deleting files that 
+     * shouldn't get compiled by the CLI. Generally, we delete all CMake artifacts and multithreaded
+     * support files (including semaphores and thread folders)
+     *
+     * @param dir The folder to search for folders and files to delete. 
+     * @throws IOException If the given folder and unneeded files cannot be deleted.
+     */
+    public static void arduinoDeleteHelper(Path dir) throws IOException {
+        deleteDirectory(dir.resolve("core/federated")); // TODO: Add Federated Support to Arduino
+        deleteDirectory(dir.resolve("include/core/federated")); // TODO: Add Federated Support to Arduino
+        
+        deleteDirectory(dir.resolve("core/threaded")); // No Threaded Support for Arduino
+        deleteDirectory(dir.resolve("include/core/threaded")); // No Threaded Support for Arduino
+
+        List<Path> allPaths = Files.walk(dir)
+                    .sorted(Comparator.reverseOrder())
+                    .collect(Collectors.toList());
+        for (Path path : allPaths) {
+            String toCheck = path.toString().toLowerCase();
+            if (toCheck.contains("cmake") || toCheck.contains("semaphore")
+            || path.endsWith("lib/util.c") || (toCheck.contains("core/platform/") && !toCheck.contains("lf_arduino"))) {
+                Files.delete(path);
+            } 
+        }
+    }
+
+    private static String fileNameMatchConverter(String fileName, Path currPath, Map<String, Path> fileStringToFilePath) 
+        throws NullPointerException {
+        // First get the child file
+        int lastPath = fileName.lastIndexOf(File.separator);
+        if (lastPath != -1){
+            fileName = fileName.substring(lastPath+1);
+        }
+        Path p = fileStringToFilePath.get(fileName);
+        if(p == null){
+            return "#include \"" + fileName + "\"";
+        }
+        String relativePath = currPath.getParent().relativize(p).toString();
+        return "#include \"" + relativePath + "\"";
+    }
+
+    /**
+     * Converts all includes recursively inside files within a specified folder to relative links
+     *
+     * @param dir The folder to search for includes to change. 
+     * @throws IOException If the given set of files cannot be relativized.
+     */
+    public static void relativeIncludeHelper(Path dir) throws IOException {
+        System.out.println("Relativizing all includes in " + dir.toString());
+        List<Path> allPaths = Files.walk(dir)
+            .filter(Files::isRegularFile)
+            .sorted(Comparator.reverseOrder())
+            .collect(Collectors.toList());
+        Map<String, Path> fileStringToFilePath = new HashMap<String, Path>();
+        for (Path path : allPaths) {
+            String fileName = path.getFileName().toString();
+            fileStringToFilePath.put(fileName, path);
```

Comment:
Maybe check here whether the key already exists, in which case we've hit an error condition.

Reply:
Will do.